src/EditorFeatures/CSharpTest/MakeStructReadOnly/MakeStructReadOnlyTests.cs

```diff
@@ -0,0 +1,691 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.MakeStructReadOnly;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.MakeStructReadOnly;
+
+using VerifyCS = CSharpCodeFixVerifier<
+    CSharpMakeStructReadOnlyDiagnosticAnalyzer,
+    CSharpMakeStructReadOnlyCodeFixProvider>;
+
+public class MakeStructReadOnlyTests
+{
+    private static Task TestMissingAsync(string testCode, LanguageVersion version = LanguageVersion.Preview)
+        => TestAsync(testCode, testCode, version);
+
+    private static async Task TestAsync(string testCode, string fixedCode, LanguageVersion version = LanguageVersion.Preview)
+    {
+        await new VerifyCS.Test
+        {
+            TestCode = testCode,
+            FixedCode = fixedCode,
+            LanguageVersion = version,
+        }.RunAsync();
+    }
+
+    [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeStructReadOnly)]
```

Comment:
Consider moving `Trait` from all test methods to the class.

Reply:
Done.